mapeditor/campaigneditor/scenarioproperties.cpp

```diff
@@ -0,0 +1,529 @@
+	else
+		ui->lineEditScenarioName->setStyleSheet(""); 
+
+	ui->pushButtonExport->setEnabled(map != nullptr);
+	ui->pushButtonRemove->setEnabled(map != nullptr);
+
+	ui->radioButtonStartingOptionHeroCrossover->setEnabled(scenario.getNum() > 0);
+	bool allowSelectingStartHero = false;
+	if(map != nullptr)
+		for(auto & player : map->players)
+			if(player.generateHeroAtMainTown)
+				allowSelectingStartHero = true;
+	ui->radioButtonStartingOptionStartingHero->setEnabled(allowSelectingStartHero);
+
+	ui->tabPrologueEpilogue->setEnabled(map != nullptr);
+	ui->tabCrossover->setEnabled(map != nullptr);
+	ui->tabStarting->setEnabled(map != nullptr);
+
+	ui->comboBoxDefaultDifficulty->setEnabled(map != nullptr);
+	ui->listWidgetPrerequisites->setEnabled(map != nullptr);
+	ui->plainTextEditRightClickText->setEnabled(map != nullptr);
+
+	ui->tableWidgetStartingCrossover->clearContents();
+	ui->tableWidgetStartingCrossover->setRowCount(0);
+	ui->listWidgetStartingBonusOption->clear();
+
+	if(map != nullptr)
+	{
+		std::vector<PlayerColor> selectableColors;
+		for(int i = 0; i < map->players.size(); i++)
+			if(map->players[i].canHumanPlay)
+				selectableColors.push_back(PlayerColor(i));
+
+		ui->comboBoxStartingBonusPlayerPosition->clear();
+		for(int i = 0; i < PlayerColor::PLAYER_LIMIT_I; ++i)
+		{
+			if(!vstd::contains(selectableColors, PlayerColor(i)))
+				continue;
+
+			MetaString str;
+			str.appendRawString((tr("Player") + " ").toStdString());
+			str.appendNumber(i + 1);
+			str.appendRawString(" (");
+			str.appendName(PlayerColor(i));
+			str.appendRawString(")");
+			ui->comboBoxStartingBonusPlayerPosition->addItem(QString::fromStdString(str.toString()), QVariant(i));
+		}
+		int index = ui->comboBoxStartingBonusPlayerPosition->findData(campaignState->scenarios.at(scenario).travelOptions.playerColor.getNum());
+		if(index != -1)
+			ui->comboBoxStartingBonusPlayerPosition->setCurrentIndex(index);
+	}
+
+	if(campaignState->scenarios.at(scenario).travelOptions.startOptions == CampaignStartOptions::HERO_CROSSOVER || campaignState->scenarios.at(scenario).travelOptions.startOptions == CampaignStartOptions::HERO_OPTIONS)
+	{
+		if(campaignState->scenarios.at(scenario).travelOptions.startOptions == CampaignStartOptions::HERO_OPTIONS)
+			on_radioButtonStartingOptionStartingHero_toggled();
+		else
+			on_radioButtonStartingOptionHeroCrossover_toggled();
+
+		for(int i = 0; i < campaignState->scenarios.at(scenario).travelOptions.bonusesToChoose.size(); i++)
+		{
+			auto bonus = campaignState->scenarios.at(scenario).travelOptions.bonusesToChoose[i];
+
+			ui->tableWidgetStartingCrossover->insertRow(i);
+			QComboBox* comboBoxOption = new QComboBox();
+			if(campaignState->scenarios.at(scenario).travelOptions.startOptions == CampaignStartOptions::HERO_OPTIONS)
+			{
+				for(auto & hero : heroSelection)
+					comboBoxOption->addItem(QString::fromStdString(hero.second), hero.first);
+			}
+			else
+			{
+				for(int i = 0; i < scenario.getNum(); ++i)
```

Comment:
Both loops use `i` for variable name.

Technically legal, but please rename inner loop variable to `j` and adjust code